File: packages/conversation/src/agent/link.ts
/**
 * 这条连接此刻的链路态。
 *
 * 它是一帧的载荷（frame.rs 的 RunFrame::LinkChanged），所以重放一条对话会原样
 * 再演一遍。判别式与字段名与原生侧的 LinkState 逐字相同
 * （crates/kap-client/src/link.rs）。
 */
export type SessionLink =
  | {
      readonly state: 'retrying'
      readonly attempt: number
      readonly of: number
      /** 下一次重连的时刻（epoch 毫秒）；等于此刻表示正在拨号。 */
      readonly retryAt: number
      readonly reason: string
    }
  | { readonly state: 'recovered'; readonly reason: string }
  | { readonly state: 'severed'; readonly attempts: number; readonly reason: string }

File: crates/conversation/src/link.rs
//! 连接链路态，只有状态：重连策略住在 kap-client 的 link.rs；「模型半天不说话」是轮次的事，由轮次封条表达，不从这里冒充断线。

use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
#[serde(
    tag = "state",
    rename_all = "snake_case",
    rename_all_fields = "camelCase"
)]
pub enum LinkState {
    Retrying {
        attempt: u32,
        of: u32,
        retry_at: i64,
        reason: String,
    },
    Recovered { reason: String },
    Severed { attempts: u32, reason: String },
}
